### PR TITLE
[#1272] Hardened XML submission export to handle multi-value fields

### LINKED ISSUE #1272: Download as XML throws a 500-error
Via Sentry 301752

```
TypeError: Argument must be bytes or unicode, got 'list'
  File "django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/contrib/admin/options.py", line 606, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "django/utils/decorators.py", line 142, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "django/contrib/admin/sites.py", line 223, in inner
    return view(request, *args, **kwargs)
  File "django/utils/decorators.py", line 45, in _wrapper
    return bound_method(*args, **kwargs)
  File "django/utils/decorators.py", line 142, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "django/contrib/admin/options.py", line 1711, in changelist_view
    response = self.response_action(request, queryset=cl.get_queryset(request))
  File "django/contrib/admin/options.py", line 1397, in response_action
    response = func(self, request, queryset)
  File "openforms/submissions/admin.py", line 270, in export_xml
    return self._export(request, queryset, ExportFileTypes.XML)
  File "openforms/submissions/admin.py", line 246, in _export
    return export_submissions(queryset, file_type)
  File "openforms/submissions/exports.py", line 56, in export_submissions
    export_data.export(file_type.extension),
  File "tablib/core.py", line 427, in export
    return fmt.export_set(self, **kwargs)
  File "openforms/submissions/exports.py", line 76, in export_set
    field.text = serialize_objects_handler(value)
  File "src/lxml/etree.pyx", line 1039, in lxml.etree._Element.text.__set__
  File "src/lxml/apihelpers.pxi", line 747, in lxml.etree._setNodeText
  File "src/lxml/apihelpers.pxi", line 735, in lxml.etree._createTextNode
  File "src/lxml/apihelpers.pxi", line 1538, in lxml.etree._utf8
```

### PR DESCRIPTION
**Pull request by Bartvaderkin**
Fixes #1272

### CODE REVIEW

**Comment by Bartvaderkin** [comment]
I think the export value formatting should maybe also go through the formatters app, since its field type dependent formatting/serialising.

**Comment by sergei-maertens** [comment]
> I think the export value formatting should maybe also go through the formatters app, since its field type dependent formatting/serializing but that's maybe for another time.

Agree - and it would tie in with the Camunda registration backend which also requires Formio components to be converted into python types: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/registrations/contrib/camunda/type_mapping.py

All this kind of functionality really belongs in the `openforms.formio` package.

File: src/openforms/submissions/exports.py [hunk 1]
```diff
@@ -61,6 +62,13 @@ def export_submissions(
     return response
 
 
+def xml_serialize_value(value) -> str:
+    if isinstance(value, (list, dict)):
+        return json.dumps(value)
```
Comment (sergei-maertens) [inline src/openforms/submissions/exports.py:67]: I don't like that suddenly we have JSON inside of XML and require XML parsers to also parse JSON.

For an export with singular values, we now get this:

```xml
<submissions>
  <submission>
    <field name="Formuliernaam">Camunda demo sprint 11</field>
    <field name="Inzendingdatum">2022-01-17T15:18:02.140744</field>
    <field name="bedrag">500</field>
    <field name="invoiceCategory">Misc</field>
  </submission>
</submissions>
```

which is okay. But the multiple variant would result in:

```xml
<submissions>
  <submission>
    <field name="Formuliernaam">Camunda demo sprint 11</field>
    <field name="Inzendingdatum">2022-01-17T15:18:02.140744</field>
    <field name="bedrag">[500, 600]</field>
    <field name="invoiceCategory">["Misc", "Staff"]</field>
  </submission>
</submissions>
```

how does a parser determine what to parse as JSON and what to parse as string?

I'd much prefer the following output which sticks to XML semantics:

```xml
<submissions>
  <submission>
    <field name="Formuliernaam"><value>Camunda demo sprint 11</value></field>
    <field name="Inzendingdatum"><value>2022-01-17T15:18:02.140744</value></field>
    <field name="bedrag">
        <value>500</value>
        <value>600</value>
    </field>
    <field name="invoiceCategory">
        <value>Misc</value>
        <value>Staff</value>
    </field>
  </submission>
</submissions>
```

* this normalizes that `field` has at least one `value` element
* this opens up the path to add type information in the future / multi information: e.g. `<field name="bedrag" type="int" multiple="true">`

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/submissions/exports.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import dataclasses
+import json
```
Comment (sergei-maertens) [inline src/openforms/submissions/exports.py:2]:
```suggestion
```

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**